src/components/Result/Result.js

```diff
@@ -15,7 +15,9 @@ class Result extends Component {
   //Algorithm to calculate the share for each person
   componentDidMount = () => {
     this.props.isNotParticipantList();
+    console.log(this.props.participants);
     var dataTemp = this.props.participants;
+    dataTemp.shift();
```

Comment:
Could you elaborate why we `shift` the array here?  Also, on the line above (no.19), let's use spread operator to copy value from `this.props.participants` to `dataTemp`. Using `=` doesn't assign an array by reference to memory, not by value => if `dataTemp` changes, `this.props.participants` changes too.

Reply:
Since right now in the participantList reducer, when the user wants to add a new participant, there will be an empty object initially added to the participants array in the state, therefore when calculating the share for each participant, there will be errors since this object does not contain any information. By using `array.shift()`, we make sure that the first element in the array (aka that empty object) will be removed from the array, ensuring that the final result is correct.

About the spread operator, that is fixed.